### ISSUE TITLE
inference speed vs. number of classes

### ISSUE THREAD

**Issue by MasIgor** [ultralytics/yolov5#8455]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi all!

It was my believe that a given network size, always takes the same amount of time to execute more or less, indipendently from the number of classes. 

I thought that it takes longer to train a network with more classes and that the detection is becomes less stable with more classes since there are more and more possibilities to detect the wrong one, but since the size of the netwrok stays the same, i thought that the inference time remains the same.

Recently i had a network doing inference on a CPU in 90 ms. it had just a few classes, i think it was 5. These classes then became more, about 75 or so. This network took consistently 140 ms to do inference. 

Could you please explain why this is happening in layman terms? isnt the number of calculations always the same?

Thank you

best regards

**Comment by sezer-muhammed** [comment]
Hi, as far as I know num of calculations is not the same. Until the last layers the calculation numbers are same yes, but the numbers increase on the last layers to create prediction for all classes. The increased number depends on the model.
You can also see this effect models like SSD-mobilenet (On jetson nano 80 classes: 30 FPS, 1 class: 43 FPS)

In short: Convolutions etc same but prediction layer changes.

Not sure this is what I know

**Comment by MasIgor** [comment]
Thank you.. if I understood it correctly then it makes perfect sense.. so the connection of the last hidden layer to all output layers varies based on the number of output layers.. right?

**Comment by sezer-muhammed** [comment]
> Thank you.. if I understood it correctly then it makes perfect sense.. so the connection of the last hidden layer to all output layers varies based on the number of output layers.. right?

Not sure if its "only" last layer, generally there are two stages, one of them feature extraction (flatten layer). This layer only find features. Constant Output. 

The other stages predict classes based on these features, this stage can have many layers and this layers based on class number.

But this is general method. So I dont know if its also same for yolov5 (probably not same since as I know yolov5 has different approach)

But, yes some of last layers are bigger if there are more classes.

**Comment by glenn-jocher** [comment]
NMS time will also increase proportional to class count since this is a per class op.
